homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -60,33 +71,73 @@ def _show_config_form(self, user_input=None, errors=None) -> FlowResult:
             errors=errors,
         )
 
-    async def _async_is_server_online(self, address: str) -> bool:
-        """Check server connection using a 'status' request and return result."""
+    async def _async_is_java_server_online(self, address: str) -> bool:
+        """Check Java Edition server connection."""
 
         # Parse and check server address.
         try:
             server = await JavaServer.async_lookup(address)
         except ValueError as error:
             _LOGGER.debug(
                 (
-                    "Error occurred while parsing server address '%s' -"
+                    "Error occurred while parsing Java Edition server address '%s' -"
+                    " ValueError: %s"
+                ),
+                address,
+                error,
+            )
+            return False
+
+        _LOGGER.debug("Java Edition server address '%s' is valid", address)
+
+        return await self._async_is_server_online(server)
+
+    async def _async_is_bedrock_server_online(self, address: str) -> bool:
+        """Check Bedrock Edition server connection."""
+
+        # Parse and check server address.
+        try:
+            server = BedrockServer.lookup(address)
```

Comment:
bedrock isnt async? Should it run in the executor?

Reply:
The lookup function isn't async. But you're right, I forgot to run it in the executor :+1: